### ISSUE TITLE
[BUG] Class not found after using new update / install scheme

### ISSUE THREAD

**Issue by madskjeldgaard** [davidgranstrom/scnvim#99]
**Describe the bug**

After updating to the new installation scheme, SC cannot find the SCNvim classes anymore and I get "class not defined errors". I tried uninstalling and remove all old symlinks, it did not help.
After reinstalling with the `do`-hook and calling the new installation function, I get a strange symlink in my extensions folder which seems to refer to itself:

`lrwxrwxrwx  1 mads users      13 Jun 11 15:07 scide_scnvim -> /scide_scnvim`

**Expected behavior**

**Steps to reproduce**

**Additional context**

**Information**
- `nvim --version`:
- Operating system:
- Running scnvim in a Terminal/GUI:

**Comment by madskjeldgaard** [comment]
I tried completely removing the plugin using PlugClean and then reinstalling from scratch using `Plug 'davidgranstrom/scnvim', { 'do': {-> scnvim#install() } }` but the result is the same.

**Comment by madskjeldgaard** [comment]
I fixed this by manually making the symlink for now so it isn't critical

**Comment by davidgranstrom** [comment]
@madskjeldgaard Thanks for the info! Would need to know your OS and `nvim --version` in order to debug. Also, did you try `:call scnvim#install()` (not via vim-plug) after the update?

**Comment by davidgranstrom** [comment]
Just tested the new installation method on linux with my old Thinkpad but could not reproduce what you describe. I would be interested to see the output of `:lua print(package.searchpath('scnvim', package.path))` if possible to help investigate this further.

**Comment by davidgranstrom** [comment]
@madskjeldgaard Did you get a chance to revisit this? I would still be interested in the output of the command I posted above in order to debug this. 

Another thing I forgot to ask you, did you install SuperCollider *before* installing the scnvim plugin? Since the plugin depends on that some directories exists (i.e. `SuperCollider/Extensions`) linking would fail until there is a valid path (you can always re-run the installation function at a later time when SuperCollider has been installed). When I update the README next time I will make sure to mention this.

**Comment by X2theL** [comment]
I had the same issue as @madskjeldgaard `:call scnvim#install()` would not create a correct symlink. I also fixed this by creating the symlink manually. I think the automatically created symlink just pointed at /scide_scnvim or something like that.

I'm on Archlinux, nvim v0.4.4. 
`:lua print(package.searchpath('scnvim', package.path))` returns
`/home/x2thel/.local/share/nvim/plugged/scnvim//lua/scnvim.lua`

I hope this helps and thanks a lot for this great plugin!

**Comment by davidgranstrom** [comment]
@X2theL Thanks for the info! The double `/` in your output looks interesting, it is not present on my system. I will take another look at the path handling to see if this might be the problem.

**Comment by davidgranstrom** [comment]
@X2theL Also forgot to ask, did you already have an existing SuperCollider installation before trying to install scnvim?

**Comment by X2theL** [comment]
Yes, SC was previously installed via the Arch Community repo.

**Comment by madskjeldgaard** [comment]
Very sorry for the late reply - I totally forgot this. On mine, the command above spits out:
`/home/mads/.vim/plugged/scnvim/lua/scnvim.lua`
And when I now removed the plugin, manually deleted the symlink from before, everything is good. I'm on SC 3.11 now, neovim 0.4.4. I also updated to the git version of pynvim (which contained problems causing crashes with ultisnips, but that's a different issue) if that is related somehow. But anyway, it seems to work perfectly.
